lib/net.js

```diff
@@ -1065,7 +1071,10 @@ function lookupAndConnect(self, options) {
   }
 
   if (options.lookup && typeof options.lookup !== 'function')
-    throw new TypeError('"lookup" option should be a function');
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
```

Comment:
This should likely be `ERR_INVALID_OPTION_VALUE`

Reply:
There are thrown errors for all the following 4 values in `lib/net` `lookupAndConnect` function :
```
function lookupAndConnect(self, options) {
  // ...
  options.port
  options.localAddress
   options.localPort
  options.lookup
  // ...
```
So  I guess that for consistency, all the errors thrown for invalid values of the above vars should have a code of `ERR_INVALID_OPTION_VALUE`

Reply:
I checked how we use mainly use this right now in the code base and it seems like `ERR_INVALID_ARG_TYPE` is the right type here. We might improve those types in general later on further but I think this should not block the PR from being merged.